components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  Component,
+  OnInit,
+  ViewChild,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+
+import { FormBuilder, Validators, FormGroup } from '@angular/forms';
+import { Regex } from 'app/helpers/auth/regex';
+
+import { MatSelect } from '@angular/material/select';
+import { DynamicGrid } from '../create-environment-modal/create-environment-modal.component';
+
+@Component({
+  selector: 'app-infra-environment-constraint',
+  templateUrl: './infra-environment-constraint.component.html',
+  styleUrls: ['./infra-environment-constraint.component.scss']
+})
+
+export class InfraEnvironmentConstraintComponent implements OnInit {
+
+  @Input() constraintKeys: string[] = [];
+  @Input() dynamicArray: Array<DynamicGrid> = [];
+  @Input() name_id: string;
+  @Output() constraintValues: EventEmitter<DynamicGrid[]> = new EventEmitter();
+  @ViewChild(MatSelect) select: MatSelect;
+
+  public conflictError = false;
+  public operator_id = '';
+  public operatorKeys: string[] = [];
+  public secondFormGroup: FormGroup;
+  public selectedLangs: string[] = [];
+  public rowKeys: string[] = [];
+
+  constructor(
+    private fb: FormBuilder
+  ) {
+    this.secondFormGroup = this.fb.group({
+      version: ['', [Validators.required, Validators.pattern(Regex.patterns.VALID_VERSION)]]
+    });
+
+    this.operatorKeys = ['~>', '>=', '>', '=', '<', '<='];
```

Comment:
Add an in-code comment explaining `~>`

Reply:
added code comments.

Reply:
Again, you did not read my request. 
(1) Please delete the comment `// operators for Cookbook version`; that is self-evident.
(2) Please add an in-code comment explaining `~>`

Reply:
Added in-code comment for ```~>``` :  **```approximately greater than; also known as "pessimistically greater than", or "pessimistic"```**

Reply:
Are you looking for this above change?